File: GreatCircle.java
public class GreatCircle {
    public static void main(String[] args) {
        double x1 = Double.parseDouble(args[0]);
        double y1 = Double.parseDouble(args[1]);
        double x2 = Double.parseDouble(args[2]);
        double y2 = Double.parseDouble(args[3]);
        double x1InRad = Math.toRadians(x1);
        double y1InRad = Math.toRadians(y1);
        double x2InRad = Math.toRadians(x2);
        double y2InRad = Math.toRadians(y2);
        double sinX = Math.sin((x2InRad - x1InRad) / 2);
        double sinY = Math.sin((y2InRad - y1InRad) / 2);
        double sinXSquared = Math.pow(sinX, 2);
        double sinYSquared = Math.pow(sinY, 2);
        double formulaInBrackets = Math.sqrt(sinXSquared + Math.cos(x1InRad) * Math.cos(x2InRad) * sinYSquared);
        double arcSinFormula = Math.asin(formulaInBrackets);
        double r = 6371.0;
        double fullFormula = 2 * r * arcSinFormula;
        System.out.println(fullFormula + " kilometers");
    }
}

File: GreatCircle.py
import math


def great_circle(x1, y1, x2, y2):
    x1 = math.radians(x1)
    y1 = math.radians(y1)
    x2 = math.radians(x2)
    y2 = math.radians(y2)
    r = 6371.0
    sin_x = math.sin((x2-x1) / 2)
    sin_y = math.sin((y2-y1) / 2)
    pow_sin_x = math.pow(sin_x, 2)
    pow_sin_y = math.pow(sin_y, 2)
    formula_in_bracket = math.sqrt(
        pow_sin_x + math.cos(x1) * math.cos(x2) * pow_sin_y)
    formula = 2 * r * math.asin(formula_in_bracket)
    return f"{formula} kilometers"


print(great_circle(60.0, 15.0, 120.0, 105.0))
